test/cpp/api/sequential.cpp

```diff
@@ -29,11 +29,11 @@ TEST_F(SequentialTest, ConstructsFromSharedPointer) {
       return value;
     }
   };
-  Sequential sequential(
+  AnySequential sequential(
```

Comment:
I think we should do a few things to this test suite:
1. Change all sequential in this file to output a Tensor, and then construct them as `Sequential` instead of `AnySequential`.
2. Add a test that showcase the usage of `AnySequential` when users need to output a non-Tensor type.
3. Add a test that shows `Sequential` can be nested, while `AnySequential` cannot. We should use `ASSERT_THROWS_WITH` to catch the `AnySequential` nesting failure in the test.
4. Add a test that shows using `Sequential->forward<float>(...)` is no longer supported, and test that it throws a nice error message encouraging people to switch to `AnySequential`.

The reason is that `Sequential` is the class we want to encourage people to use (as it can be nested), while they should use `AnySequential` only when they have to output a non-Tensor type.

Reply:
@yf225 We can't do ```ASSERT_THROWS_WITH``` because the problem is a ```static_assert``` and another compilation error about ```make_holder```.